kratos/mpi/includes/mpi_data_communicator.h

```diff
@@ -383,7 +399,8 @@ class KRATOS_API(KRATOS_MPI_CORE) MPIDataCommunicator: public DataCommunicator
     ///@name Member Variables
     ///@{
 
-    MPI_Comm mComm;
+    MPI_Comm mComm;                      /// The MPI communicator
+    mutable std::vector<int> mRanksList; /// The list of ranks involved in this DataCommunicator
```

Comment:
why `mutable`?

Reply:
also, the doxygen annotations should precede the item they refer to:
```suggestion
    /// The list of ranks involved in this DataCommunicator
    mutable std::vector<int> mRanksList;
```

Reply:
Mutable as it is modified when const, but with the idea of @matekelemen can be avoided